tests/test_optimize.py

```diff
@@ -54,25 +54,25 @@
 univariate_funcs = [np.sin,
                     lambda x: np.exp(x / 10.),
                     lambda x: x ** 2]
-grad_uni_fns = [np.cos,
-                lambda x: np.exp(x / 10.) / 10.,
-                lambda x: 2 * x]
+grad_uni_fns = [lambda x: (np.cos(x), ),
+                lambda x: (np.exp(x / 10.) / 10., ),
+                lambda x: (2 * x, )]
```

Comment:
Just to confirm I'm following this change:

* This is test data for optimizers being passed custom gradient functions (via the `grad_fn` argument)

* We now need to make sure that these custom gradient functions conform to the same output type/shape as `qml.grad`, which will always return a tuple rather than a scalar

I just ran this locally and can confirm this behaviour:

```python
In [4]: qml.grad(np.cos)(0.5)
Out[4]: (array(-0.47942554),)
```

Out of curiosity, how come these tests were passing before? Was it a bug in the existing optimizer tests?

Reply:
`flatten` removed all of the nesting.  `qml.grad` and the manual functions would flatten to the same thing.  

Now though, I'm relying on having something I can iterate over the same way I'm iterating over arguments.